src/Operation/InsertOne.php

```diff
@@ -103,7 +105,14 @@ public function execute(Server $server)
         $insertedId = $bulk->insert($this->document);
 
         $writeConcern = isset($this->options['writeConcern']) ? $this->options['writeConcern'] : null;
-        $writeResult = $server->executeBulkWrite($this->databaseName . '.' . $this->collectionName, $bulk, $writeConcern);
+        try {
+            $writeResult = $server->executeBulkWrite($this->databaseName . '.' . $this->collectionName, $bulk, $writeConcern);
+            return new InsertOneResult($writeResult, $insertedId);
+       } catch (BulkWriteException $e) {
+            if ($e->getCode() == "E11000") {
```

Comment:
"E11000" appears in the error message, but the exception code is an integer.

It looks like there are several codes to be handled (see: [`_is_duplicate_key_error()`](https://github.com/mongodb/mongo-c-driver/blob/1.8.1/src/mongoc/mongoc-write-command.c#L62) in libmongoc). We'll soon be able to drop support for MongoDB 2.6, but that won't be until PHPLIB 1.3.0 (for the MongoDB 3.6-compatible release). Even so, the odd mongos code for 16460 will need to be handled (from [SERVER-11493](jira.mongodb.org/browse/SERVER-11493)). While libmongoc only checks the integer code, the server has a more robust check that also checks the error message (see: [this code search](https://github.com/mongodb/mongo/search?utf8=%E2%9C%93&q=16460&type=)).

Additionally, it looks like PHPC tends to leave the error code unset as zero when throwing a BulkWriteException in [`phongo_execute_write()`](https://github.com/mongodb/mongo-php-driver/blob/1.3.1/php_phongo.c#L503). This means we'll actually need to dig into the WriteResult object within BulkWriteException to inspect it for errors.